### PR TITLE
[WIP] Form reset mixin

### PR DESCRIPTION
**Pull request by scottgonzalez**
Everything is thrown together right now, but I wanted to get some discussion about the tests. Is this good enough or are there more/different things that should be tested?

Note: This is built on top of gh-1554.

### CODE REVIEW

File: tests/unit/form-reset-mixin/core.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			refresh: function() {
+				$.ui.testWidget.refreshed.push( this.element.attr( "id" ) );
+			}
+		} ] );
+
+		$.ui.testWidget.refreshed = [];
+	},
+
+	teardown: function() {
+		if ( $.ui ) {
```
Comment (jzaefferer) [inline tests/unit/form-reset-mixin/core.js]: Is this `if` actually needed? Looks like `$.ui` these will always be defined.

**Comment by jzaefferer** [comment]
This looks good to me. I considered checking that the data actually references the right instances, but the `$.ui.testWidget.refreshed` assertion implies that already. Can't think of anything else to test.

File: tests/unit/form-reset-mixin/core.js (hunk 1 above)
Comment (scottgonzalez) [inline tests/unit/form-reset-mixin/core.js]: Yeah, it's always defined. I had copied this from the widget tests, where it may not exist yet.

**Comment by scottgonzalez** [comment]
Ok, I'll remove that one conditional in the tests and land this and gh-1554.

**Pull request closed without merging**